### PR TITLE
Test all supported versions of node

### PR DESCRIPTION
**Pull request by stephenmichaelf**
#42 

This change adds a way for us to test all supported node versions locally.

### CODE REVIEW

**Comment by bryanmacfarlane** [comment]
Works fine on windows?  I didn't think nvm worked on windows??

**Comment by stephenmichaelf** [comment]
Yes sir, works on Windows. I use [this one](https://github.com/coreybutler/nvm-windows) which I believe functions the same as the non-windows version. Did the tests work on Mac?

File: make.js [hunk 1]
```diff
@@ -71,6 +75,77 @@ target.test = function() {
...
 }
 
+target.testall = function() {
+    // make sure we have nvm
+    var nvmCommand = runCommand('nvm');
+    if (nvmCommand.indexOf('Running version') === -1) {
+        fail('requires nvm to be installed');
+    }
+    
+    var versionsToTest = ['8.9.1', '6.12.0', '4.8.6'];
```
Comment (bryanmacfarlane) [inline make.js]: This is already out of date.  6.12.3 is latest LTS.
Why not simplify all this logic here and simply use, 4, 6, 8
Then run nvm install x and nvm use x everytime.

The only time nvm install will take a time hit is when there's a new version and that's what we want.

Comment (stephenmichaelf) [reply make.js]: According to [this](https://stackoverflow.com/questions/44789330/use-nvm-to-install-latest-nodejs-for-specific-major-version) we should be able to say "nvm install 8" and get the latest LTS but it's not working, at least for the Windows version. I am guessing there are differences in the versions but not sure.

One alternative would be to do "nvm list available" and parse that, but it seems like overkill. We could also make a REST call I am guessing but that too seems like overkill. I will keep looking into this.

Comment (stephenmichaelf) [reply make.js]: After looking [here](https://github.com/coreybutler/nvm-windows/issues/156) it looks like the Windows version behaves differently, I requested a change but looks like we will need a workaround that works cross plat.

Comment (stephenmichaelf) [reply make.js]: Discussed offline, plan is to make it work for Linux/OSX and the command will do nothing on Windows.

**Comment by stephenmichaelf** [comment]
Discussed offline, we are removing this feature.

**Pull request closed without merging**